packages/api/tests/endpoints/test-rules.js

```diff
@@ -390,16 +536,82 @@ test.serial('POST returns a 500 response if record creation throws unexpected er
   }
 });
 
+test.serial('POST does not write to RDS or Dynamo if writing to DynamoDB fails', async (t) => {
+  const { dbClient } = t.context;
+  const newRule = fakeRuleFactoryV2();
+  newRule.provider = undefined;
+  newRule.collection = undefined;
+
+  const failingDbClient = {
+    insert: () => {
+      throw new Error('Insert Rule Error');
+    },
+  };
+
+  const expressRequest = {
+    body: newRule,
+    testContext: {
+      dbClient: failingDbClient,
+    },
+  };
+
+  const response = buildFakeExpressResponse();
+
+  await post(expressRequest, response);
+  const dbRecords = await dbClient.select()
+    .from(tableNames.rules)
+    .where({ name: newRule.name });
+
+  t.false(await ruleModel.exists(newRule.name));
```

Comment:
This is interesting. I wouldn't expect this to pass based on my understanding of how the transactions work. Let me take a closer look

Reply:
Ah, so this test is passing because your faked DB client is missing a `transaction` function, which is now required by the endpoint code. I'd run this test through the debugger and inspect the thrown error to see what I mean

Reply:
[`717d0e5` (#2012)](https://github.com/nasa/cumulus/pull/2012/commits/717d0e50dc816dd503b437d92a583a563a6c0766)